upstream/controllers.go

```diff
@@ -0,0 +1,103 @@
+package upstream
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/flanksource/commons/logger"
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db"
+	"github.com/labstack/echo/v4"
+	"gorm.io/gorm/clause"
+)
+
+func UpstreamPushesCtrl(c echo.Context) error {
```

Comment:
maybe UpstreamPusher ? Since its in the controllers file, we shouldn't have to suffix the `Ctrl` part

Reply:
It's also part of the `upstream` pkg which is not just for controllers _(I suppose)_. So from the perspective of all the APIs that we expose from the `upstream` pkg I think it's nice to have a suffix for clarity. Unless we just have controllers in the package.

Reply:
Then its better to use a verb.context method, like `PushUpstream`